Add endpoint for latest app icon

svenstucki opened the pull request:
Adds a new endpoint `/reports/<app handle>/icon` which returns the latest icon fetched for this app. The endpoint is used in the search results to fix #223.

I hope this is along the lines of what you expect. Just let me know otherwise :).

Potentially `/reports/<app handle>/latest/icon` would be a better endpoint or maybe move it into the REST API - the current endpoint is in the i18n patterns.

svenstucki committed:
Add endpoint for latest app icon

pnu-s commented:
That's awesome, thanks a lot @svenstucki! :)

Everything looks good to me and it works as expected.

I have just 2 minor comments/questions:
- I agree that `/reports/<app handle>/latest/icon` would make more sense, except if you had a particular reason to prefer `/reports/<app handle>/icon`
- Out of curiosity, why did you change the name of the function from `get_app_icon()` to simply `icon()`?

svenstucki commented:
Thanks for the feedback, updated the PR.

- Changed to this path now :).
- Honestly, not sure anymore. I like the concise names and probably thought it would fit in better with the `index`. But when looking at it again today, it's really not warranted compared to the other names. Changed it back.

pnu-s commented:
Great, thanks again! :)

(To be fair, I don't know neither which solution was better for the function name)

svenstucki commented:
Thank you!
